### PR TITLE
🧪 Add unit tests for route_decision function

### PR DESCRIPTION
**Pull request by prateekmulye**
### 🎯 What: The testing gap addressed
The `route_decision` function in `F1AgentGraph` was previously untested. This function is a critical component of the agent's orchestration logic, as it determines the next node in the LangGraph state machine based on query analysis.

### 📊 Coverage: What scenarios are now tested
The new test suite in `tests/agent/test_graph.py` covers the following scenarios:
- **Off-topic queries**: Verifies that queries with `off_topic` intent are routed directly to generation, bypassing retrieval.
- **Combined retrieval**: Verifies that queries requiring both historical and real-time data are routed to the parallel retrieval node.
- **Vector-only retrieval**: Verifies routing when only historical data (from the vector store) is needed.
- **Search-only retrieval**: Verifies routing when only real-time data (from Tavily) is needed.
- **Direct generation**: Verifies routing for general queries that don't require external retrieval.
- **Edge cases**: Handles missing routing decision metadata gracefully.

### ✨ Result: The improvement in test coverage
Adding these unit tests ensures the reliability of the agent's core routing logic and prevents regressions during future refactoring of the graph orchestration. The tests are designed to be fast, deterministic, and isolated from external dependencies.

---
*PR created automatically by Jules for task [16271734519827607092](https://jules.google.com/task/16271734519827607092) started by @prateekmulye*

### CODE REVIEW

**Comment by prateekmulye** [comment]
@jules the checks are feeling as you can see all the checks and all the PR's essentially this PR fails at when when the build is trying to execute black The the formatting tool so I need you to iteratively run and fix all the black related for bugs and then continue monitoring the the build and fix every fix everything until the builds are green

File: src/prompts/rag_prompts.py [hunk 1]
```diff
@@ -7,9 +7,12 @@
...
-                                    MessagesPlaceholder, PromptTemplate)
+from langchain_core.prompts import (
+    ChatPromptTemplate,
+    HumanMessagePromptTemplate,
+    MessagesPlaceholder,
+    PromptTemplate,
+)
 
 # RAG prompt for combining vector context with queries
 RAG_CONTEXT_TEMPLATE = """You are ChatFormula1, an expert Formula 1 analyst. Use the provided context to answer the user's question accurately.
```
Comment (prateekmulye) [inline src/prompts/rag_prompts.py]: @jules These prompts are outdated and need to be reworked. The name of our application is no longer Chat Formula 1; it is now Chat P1, and the company name is different. Also, I need a cleanup on all these prompts. Only keep the ones that are relevant. Only keep the ones that are being used. We don't need prompts that aren't being used or contradict each other. They have security leaks.

Comment (prateekmulye) [reply src/prompts/rag_prompts.py]: @jules, we must be cognizant of the fact that context loading, context mishaps, and context clearing are real problems in AI right now, so we need to find ways to optimize this context window. We must use caching and advanced prompting techniques to maintain our performance and accuracy to have a state-of-the-art AI system for ChatP1.ai. 

## Our prompt system should mimic how Anthropic is structured, or rather that they have recommended how the system should be structured and also how Gemini and their systems are structured. Our prompts should have proper guardrails, and they should be less broad but also not too targeted. Essentially, we want to find a beautiful, perfect middle ground of prompt where our context is not bloated and we are smart and decisive and efficient in routing and having our prompts for our Formula One Chat P1 system.

File: src/prompts/specialized_prompts.py [hunk 2]
```diff
@@ -7,8 +7,11 @@
...
 
 from langchain_core.output_parsers import PydanticOutputParser, StrOutputParser
-from langchain_core.prompts import (ChatPromptTemplate, FewShotPromptTemplate,
-                                    PromptTemplate)
+from langchain_core.prompts import (
+    ChatPromptTemplate,
+    FewShotPromptTemplate,
+    PromptTemplate,
+)
 from pydantic import BaseModel, Field
```
Comment (prateekmulye) [inline src/prompts/specialized_prompts.py:14]: @jules I noticed these prompts, though they are comprehensive, might be outdated for, for example, a system that is predominantly heavy on 2026 Formula 1 regulations. The fact that it is extremely smart and authoritative in its knowledge base of the differences between the old regulation and the new regulation makes it exciting and makes it everything new.

Again, as I said in the previous prompt context, bloating is important that we avoid actively. We use the proper anthropic prescribed and TWG prescribed, or anthropic prescribed, way of prompting in these situations. We want to make sure that hallucination is to a minimum and that creativity and everything have to be with the proper tone and context. We are Formula 1; we are sports media essentially, so we need to be that excited. We should have that level of tone and excitement and not a typical AI response prompt or something like that.

File: src/prompts/specialized_prompts.py
Comment (prateekmulye) [inline src/prompts/specialized_prompts.py:152]: @jules, I needed to make sure we are not including anything that is not production-ready or that does not demonstrate proper system architectural thinking based on a production-based architecture.

Comment (prateekmulye) [reply src/prompts/specialized_prompts.py:152]: @jules as I have mentioned in the previous comments as well, the season that we're building this for is 2026 and beyond, and as well all the prior seasons, but that is only for historical sense when people want to talk about the historical sense. But predominantly, it is going to be a season 2026 and forward.

Since the regulations have changed and there are new teams, new drivers, and new circuits, we should never hardcore code anything. We should rely on a dynamic scheduling of dynamic wherever there is possibility. We should use dynamic loading but use strict guidance so that the AI does not hallucinate ever and starts predicting or analyzing 2025 thinking that that is the current year. You have to emphasize whenever that is whenever current means it has to be 2026 or 2027 or whatever the current year and the current regulations are.

One other basic and most important thing: regulations do change, regulations do alter as the season progresses.

**Review by prateekmulye — COMMENTED** [review]
@jules, I've left a few highly important comments for you to analyze and systematically fix without introducing any new bugs.
